solutions/cf-1713E.mdx

```diff
@@ -0,0 +1,164 @@
+
+After this we can construct the answer by iterating over each cell of the grid. If the cell's row index is equal to its column index (i.e. it's on the diagonal of the grid) it can never change regardless of the flips. If $a[i][j] == a[j][i]$ then we can manually set them (since we didn't merge them in the disjoint set union when comparing them the states might have not been merged). Otherwise, we can check whether $i$ and $j$ are merged in the disjoint set union. If they aren't then we will swap $a[i][j]$ and $a[j][i]$. Otherwise we will keep them the same.
+
+## Implementation
+
+**Time Complexity:** $\mathcal{O}(N^2)$
+
+<LanguageSection>
+
+<CPPSection>
+
+```cpp
+#include <bits/stdc++.h>
+using namespace std;
+
+using vi = vector<int>;
+
+//BeginCodeSnip{DSU}
+struct DSU {
+	vi e;
+	DSU(int n) {
+		e = vi(n, -1);
+	}
+	
+	int get(int x) {
+		if (e[x] < 0) {
+			return x;
+		} else {
+			e[x] = get(e[x]);
+			return e[x];
+		}
+	}
+	
+	bool same(int a, int b) {
+		return get(a) == get(b);
+	}
+	
+	int size(int x) {
+		return -e[get(x)];
+	}
+	
+	bool unite(int x, int y) {
+		x = get(x);
+		y = get(y);
+		if (x == y) {
+			return false;
+		}
+		if (e[x] > e[y]) {
+			swap(x, y);
+		}
+		e[x] += e[y];
+		e[y] = x;
+		return true;
+	}
+};
+//EndCodeSnip
+
+void solve() {
+	int n;
+	cin >> n;
+	int a[n][n];
+	for (int i = 0; i < n; i++) {
+		for (int j = 0; j < n; j++) {
+			cin >> a[i][j];
+		}
+	}
+	
+	DSU dsu(2 * n);
+	
+	for (int i = 0; i < n; i++) {
+		// Only check each pair once and diagnals don't matter
+		for (int j = i + 1; j < n; j++) {
+			// If they are the same we should not merge either way
+			// because it could affect later merges
+			if (a[i][j] == a[j][i]) {
+				continue;
+			} else if (a[i][j] > a[j][i]) {
+				// Swap the elements if the states are not already merged the opposite way
+				if (dsu.get(i) != dsu.get(j)) {
+					dsu.unite(i, j + n);
+					dsu.unite(j, i + n);
+				}
+			} else	{
+				// Keep the elements the same if the states
+				// are not already merged the opposite way
+				if (dsu.get(i) != dsu.get(j + n)) {
+					dsu.unite(i, j);
+					dsu.unite(i + n, j + n);
+				}
+			}
+		}
+	}
+	
+	// Construct the answer
+	int ans[n][n];
+	for (int i = 0; i < n; i++) {
+		for (int j = i; j < n; j++) {
+			/*
+			 * If a[i][j] is part of a diagonal or the pair is equal (a[i][j] == a[j][i]),
+			 * manually set it to be the same as the original array
+			 */
+			if (i == j) {
+				ans[i][j] = a[i][j];
+			} else if (a[i][j] == a[j][i]) {
+				ans[i][j] = ans[j][i] = a[i][j];
+			} else {  // Otherwise check whether the i and j states are merged
+				if (dsu.get(i) == dsu.get(j)) {
+					ans[i][j] = a[i][j];
+					ans[j][i] = a[j][i];
+				} else {
+					ans[i][j] = a[j][i];
+					ans[j][i] = a[i][j];
+				}
+			}
+		}
+	}
+	
+	for (int i = 0; i < n; i++) {
+		for (int j = 0; j < n; j++) {
+			cout << ans[i][j] << " ";
+		}
+		cout << endl;
```

Comment:
perhaps try using `\n` here instead of `endl`, see if that fixes the vector issue

Reply:
yea it fixed the issue